### PR TITLE
Stop using Rabit in prediction

### PR DESCRIPTION
**Pull request by austinzh**
As discuss in https://github.com/dmlc/xgboost/issues/4628
[PreXGBoost.scala:transformDataset](https://github.com/dmlc/xgboost/blob/a68580e2a7d84d7206165ab1f3c911886d45dcba/jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/PreXGBoost.scala#L190) cause succession  training fail because it can not shutdown Rabit if iterator did not reach the end.

We shutdown Rabit only on the next() being call and hasNext() is false. As it show below.
```
        override def next(): Row = {
          val ret = batchIterImpl.next()
          if (!batchIterImpl.hasNext) {
            Rabit.shutdown()
          }
          ret
        }
```
But when we call show() or limit() instead of collect() or write() iterator won't reach hasNext() == false

And as for prediction, we don't need Rabit anyway, so there is no point to start it at all.
This PR is to remove it from prediction.

### CODE REVIEW

File: jvm-packages/dev/Dockerfile [hunk 1]
```diff
@@ -21,10 +21,10 @@ FROM centos:7
 # Install all basic requirements
 RUN \
     yum -y update && \
-    yum install -y bzip2 make tar unzip wget xz git centos-release-scl yum-utils java-1.8.0-openjdk-devel && \
+    yum install -y which bzip2 make tar unzip wget xz git centos-release-scl yum-utils java-1.8.0-openjdk-devel && \
     yum-config-manager --enable centos-sclo-rh-testing && \
     yum -y update && \
-    yum install -y devtoolset-7-gcc devtoolset-7-binutils devtoolset-7-gcc-c++ && \
+    yum install -y devtoolset-7-gcc devtoolset-7-binutils devtoolset-7-gcc-c++ gcc-gfortran python3-devel openblas-devel lapack-devel pkgconfig  && \
     # Python
```
Comment (trivialfis) [inline jvm-packages/dev/Dockerfile:28]: Not sure if we need to change the docker file here.

**Review by trivialfis — COMMENTED** [review]
@wbo4958 Please help take a look.

File: jvm-packages/dev/Dockerfile (hunk 1 above)
Comment (austinzh) [reply jvm-packages/dev/Dockerfile:28]: I follow the README.md in jvm-package.
./xgboost/jvm-packages/dev/build-linux.sh will fail for 
1. there is no `which` command 
2. with openblas-devel `scipy` can not be install

**Review by wbo4958 — COMMENTED** [review]
Please update the Copyright (Copyright (c) 2021-2023 by Contributors) and delete the unused import (import ml.dmlc.xgboost4j.java.Communicator)

**Comment by wbo4958** [comment]
BTW, why are you changing the docker file? if the docker file didn't work on your side, I guess, it shouldn't work on the CI/CD side either, but seems the CI/CD is working as expected?

**Comment by wbo4958** [comment]
Huh, And I checked GpuPreXGBoost transform which doesn't enable the rabit tracker. Seems I have forgotten to remove it for the CPU transform.

**Comment by trivialfis** [comment]
I think the docker file is no longer used on CI. We will delete it instead and fix the readme

**Comment by austinzh** [comment]
> I think the docker file is no longer used on CI. We will delete it instead and fix the readme

Sure, I will revert the Dockerfile then. Just keep things simple.

**Comment by trivialfis** [comment]
Please ignore the failure on Windows for now.

**Review by wbo4958 — APPROVED** [review]
LGTM. Thx

**Pull request merged**